README.md

```diff
@@ -62,12 +63,13 @@ public function render...()
     // for array
     $this->template->items = array_slice($items, $vp->getOffset(), $vp->getLength())
 }
-...
+
 protected function createComponentVisualPaginator()
 {
-    //$this->visualPaginator->setTemplatePath(__DIR__.'/VisualPaginator.latte');
-    $this->visualPaginator->setPaginatorRenderer(new BasicRenderer);
-    return $this->visualPaginator;
+    $component = $this->visualPaginator->create();
+    //$component->setTemplatePath(__DIR__.'/VisualPaginator.latte');
+    $component->setPaginatorRenderer(new BasicRenderer);
+    return $component;
 }
 ```
 
```

Comment:
I am not sure what is usage of this "or" block and if this should be edited for usages with component factory. It seems to me that in current version of Nette this does not work. Method "createComponent" needs to have optional string as first parameter (see https://github.com/nette/component-model/blob/master/src/ComponentModel/Container.php#L122 and https://github.com/nette/component-model/blob/master/src/ComponentModel/Container.php#L39).

Maybe example with callback could be added to previous example and this block could be removed from readme.

Reply:
work is completly ok, but you must use component: [nette-autowired](https://github.com/geniv/nette-autowired) but nette does not autowired component logic